src/server/pfs/server/driver_new_storage.go

```diff
@@ -158,35 +154,31 @@ func serializeShard(shard *pfs.Shard) (*types.Any, error) {
 	}, nil
 }
 
-func deserialize(compactionAny, shardAny *types.Any) (*pfs.Compaction, *pfs.Shard, error) {
-	compaction := &pfs.Compaction{}
-	if err := types.UnmarshalAny(compactionAny, compaction); err != nil {
-		return nil, nil, err
-	}
+func deserializeShard(shardAny *types.Any) (*pfs.Shard, error) {
 	shard := &pfs.Shard{}
 	if err := types.UnmarshalAny(shardAny, shard); err != nil {
-		return nil, nil, err
+		return nil, err
 	}
-	return compaction, shard, nil
+	return shard, nil
 }
 
 // (bryce) it might potentially make sense to exit if an error occurs in this function
 // because each pachd instance that errors here will lose its compaction worker without an obvious
 // notification for the user (outside of the log message).
+// (bryce) ^ maybe just a retry would be good enough.
 func (d *driver) compactionWorker() {
-	w := work.NewWorker(d.etcdClient, d.prefix, func(ctx context.Context, task, subtask *work.Task) (retErr error) {
-		compaction, shard, err := deserialize(task.Data, subtask.Data)
+	w := work.NewWorker(d.etcdClient, d.prefix, storageTaskNamespace)
+	if err := w.Run(context.Background(), func(ctx context.Context, subtask *work.Task) error {
```

Comment:
`context.Background`

Reply:
This is the correct context for a PFS compaction worker. It should always be running.